src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -57,7 +57,7 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
             var installerService = GetPackageInstallerService(document);
             var packageSources = searchNuGetPackages && symbolSearchService != null && installerService?.IsEnabled(document.Project.Id) == true
-                ? installerService.GetPackageSources()
+                ? await installerService.GetPackageSourcesAsync(allowSwitchToMainThread: false, context.CancellationToken).ConfigureAwait(false)
```

Comment:
all callers seem to pass 'false', so not sure if we actually need the parameter.  i think it's fine to say: this may not return anything in some cases.

Reply:
I prefer to have the parameter as a highlight to potentially-inconvenient behavior. If the service becomes free-threaded in the future we can eliminate the parameter.

Reply:
so the reason i don't love that is simply because we have a codepath that is effectively untested.  so i don't really know if it actually works or not :)